<!doctype html>
<html data-bs-theme="">
    <head>
                <title>MAS.885 - Alex Zhao's Public Pages</title>

            <meta charset="utf-8">
            <meta http-equiv="X-UA-Compatible" content="IE=edge">
            <meta name="viewport" content="width=device-width, initial-scale=1.0">

            
            
            

            
                <link  rel="icon" type="image/x-icon" href="../../../../../assets/favicon.ico">
            
            <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700">
            <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

            <link rel="stylesheet" href="../../../../../assets/css/bootstrap.min.css">
            <link rel="stylesheet" href="../../../../../assets/css/root.min.css">
            <link rel="stylesheet" href="../../../../../assets/css/main.min.css">
            <link rel="stylesheet" href="../../../../../assets/css/media.min.css">
            <link rel="stylesheet" href="../../../../../assets/css/mkdocstrings.min.css">
                <link href="../../../../../extra.css" rel="stylesheet">
                <script src="../../../../../extra.js"></script>
                <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
                <script src="../../../../../search/main.js"></script>

            
                
                     <style>:root {--text: var(--color-white);}</style> 
                     <style>:root {--title: var(--color-orange);}</style> 
                     <style>:root {--primary: var(--color-orange);}</style> 
                     <style>:root {--background: var(--color-black);}</style> 
                
            
    </head>

    <body>
        <div class="container py-3">
            <header>
                    <!-- block header -->
<nav class="navbar navbar-expand-xl border-bottom">
    <div class="container-fluid">
        

        
            <span class=" fs-4 title-color site-name" id="component-site-name" style="text-transform: uppercase;">Alex Zhao's Public Pages</span>
        

        <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarsMenu"
            aria-controls="navbarsMenu" aria-expanded="false" aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
        </button>

        <div class="collapse navbar-collapse flex-column ml-auto" id="navbarsMenu">
            <ul class="navbar-nav">

                <!-- block menu -->
                <li class="nav-item">
                    <!-- block menu -->
    
        <li class="nav-item" id="component-menu">
            <ul class="navbar-nav">
                        <li class="nav-item">
                            <a class="
                            nav-link text-gray text-decoration-none" href="../../../../../Public%20Pages/homepage/">[Welcome]</a>
                        </li>
                        <li class="nav-item dropdown">
                            <a class="
                            nav-link dropdown-toggle text-decoration-none" href="#" data-bs-toggle="dropdown">[Crypto]</a>
                            <ul class="dropdown-menu">
                                    <!-- block dropdown-menu -->
    <li>
        <a href="../../../../../Material Knowledge/Sciences/Coding/Crypto/Proving Systems (Plonky3, Halo2, Circle STARKs, Binius).md" class="dropdown-item text-decoration-none ">ZK Notes</a>
    </li>
<!-- endblock -->
                            </ul>
                        </li>
                        <li class="nav-item dropdown">
                            <a class="
                            nav-link dropdown-toggle text-decoration-none" href="#" data-bs-toggle="dropdown">[Coding]</a>
                            <ul class="dropdown-menu">
                                    <!-- block dropdown-menu -->
    <li>
        <a href="../../../../Sciences/Coding/Vim%20Tricks/" class="dropdown-item text-decoration-none ">Vim Tricks</a>
    </li>
<!-- endblock -->
                                    <!-- block dropdown-menu -->
    <li>
        <a href="../../../../Sciences/Coding/USACO/General%20Thoughts%2C%20Paradigms/" class="dropdown-item text-decoration-none ">USACO Notes</a>
    </li>
<!-- endblock -->
                                    <!-- block dropdown-menu -->
    <li>
        <a href="../../../../Sciences/ML/ML%20papers%20to%20read/" class="dropdown-item text-decoration-none ">ML papers</a>
    </li>
<!-- endblock -->
                            </ul>
                        </li>
                        <li class="nav-item dropdown">
                            <a class="
                            nav-link dropdown-toggle text-decoration-none" href="#" data-bs-toggle="dropdown">[Food and Cooking]</a>
                            <ul class="dropdown-menu">
                                    <!-- block dropdown-menu -->
    <li>
        <a href="../../../../Food%2BCooking/%E5%AE%B6%E5%B8%B8%E8%8F%9C%E8%B0%B1/%E7%B3%96%E9%86%8B%E6%8E%92%E9%AA%A8/" class="dropdown-item text-decoration-none ">糖醋排骨</a>
    </li>
<!-- endblock -->
                            </ul>
                        </li>
            </ul>
        </li>
<!-- endblock -->
                </li>
                <!-- endblock -->

                <!-- block search -->
                <li class="nav-item">
                    <a class="collapsed" data-bs-toggle="collapse" href="#collapseExample" role="button" aria-expanded="false" aria-controls="collapseExample">
                        <div class="md-search-icon">
                            <i class="fa fa-search" aria-hidden="true"></i>
                        </div>
                    </a>
                </li>
                <!--  endblock -->

                <!-- block source -->
                <li class="nav-item">
                    
                </li>
                <!--  endblock -->
            </ul>
        </div>
    </div>
</nav>
<!--  endblock -->
            </header>

            <main><!-- block search -->
<div class="collapse" id="collapseExample">
    <div role="search" class="search-box">
        <form id="rtd-search-form" class="wy-form" action="../../../../../search.html" method="get">
            <input type="text" name="q" class="search-query"
            placeholder="Search docs" title="Type search term here" />
        </form>
    </div>
</div>
<!-- endblock -->

                
                        <!-- block content -->
<section class="container post" id="component-content">
    <article>
        <header>
            
                <h1 class=" title" id="component-title">MAS.885</h1>
            
        </header>
        <p><p>learning to apply engineering principles to biology
- hierarchical composability
- standardized "parts"
- plug-and-play for manufacturing, etc. all kinds of stuff</p>
<h3 id="protein-design-part-1">protein design part 1</h3>
<ul>
<li>alpha helices, encouraged when gap of 3/4/7 have hydrogen bonding</li>
<li>beta-sheets, bonds form between strands connecting into a sheet, side chains alternate</li>
<li>beta-helix takes beta-sheets and it turns around an axis</li>
</ul>
<h3 id="protein-design-part-2">protein design part 2</h3>
<ul>
<li>ml / dl based!</li>
<li>there's a lot of techniques—gnn PorteinMPNN for struct2seq, RFDiffusion for func2struct, transformers (?) AlphaFold for seq2struct, MLM for protein modelling for downstream, etc.</li>
<li>have a specific protein can try to denature it with a target (ubiquitous protein cutting), generate a structure that will bind to it, etc.</li>
<li>design flow for de novo design:<ul>
<li>pick a binding site to target</li>
<li>generate thousands of de novo structures</li>
<li>generate hundreds of thousands of sequences</li>
<li>verify, filter, simulate, etc. cut down to a hundred or so</li>
</ul>
</li>
<li>space for: new autoencoder for structure? how to encode 3d structure automatically?</li>
</ul>
<h3 id="genetic-circuitsgibson-assembly">genetic circuits/gibson assembly</h3>
<p>can hack genes, viruses, etc. explore 
gibson assembly
epigenetic editors
lots of crispr variants exist (eg crispr13 for rna)
hard to get crispr to hit without creating cancers
paradigm for viral genomic editing:
1. promoters - on switches
2. capsid - delivery vehicle
3. virtal auxiliary sequences - packaging
4. delivered gene - eg crispr system for editing</p>
<h3 id="golden-gate-assembly">golden gate assembly</h3>
<p>learned about for hw! and did a whole simul in benchling
take your vector—e.g. pGGA, often used for GGA
figure out which GGA enzyme you want to use
GG enzymes cut outside of recognition site, hence you can cut the recognition sites <em>out</em> of the fragment and no self-ligation, no double-cutting
so then take the desired fragment, and design a primer that overhangs outside of said fragment with the recognition site and the matching overhang from the plasmid
then after PCR with designed primers, can GGA and ligate with plasmid in same tube, very happy!</p>
<h3 id="genome-circuits">genome circuits</h3>
<p>can utilize promoter/activation/etc. circuits (2-input AND types, e.g.) to create targeted mRNAs
Easy example is <a href="https://www.strandtx.com/strandtx-solutions/">Strand Therapeutics</a>. Self-replicating RNA drug that only activates when reacting to specific micro-RNA signatures in cells. Then it'll activate expression of {lysis, antigens, inflammation signaling} etc.
Check out paper: https://www.nature.com/articles/s41589-018-0146-9
How to create a genetic circuit that will only activate? can directly use stuff as repressor/activators. The synthetic RNA codes for binding proteins that bind to my RNA if desired compound is NOT present (they will destabilize without existence). Similarly can also code for binding proteins that bind to my RNA if desired compound IS present. Finally include replicon sequence from RNA viruses to make self-propagating copies in cells where it is expressed (i.e. target cancer cells). Am a little scared of mutations in synthetic RNA.</p>
<h3 id="synthetic-cells">synthetic cells</h3>
<p>make synthetic membranes from lipids, called cell-free systems. they can be freeze-dried, are well understood, just take the pieces that you want and mass-manufacture them with microfluidics. this gives you small easily reprogrammable biomachines. can be used for protein manufacturing, testing, color-based rna matching etc.</p>
<h3 id="biomanufacturing-ginkgo">biomanufacturing + ginkgo</h3>
<p>The goal is to gradually reduce biology to engineering. Engineering is reducing a problem to a detailed, controlled, reproducible pipeline where each stage can be iterated on in a way that should yield desired product at scale.
As in normal engineering, there are three parts:
- Design: either creating a novel protein, or taking an existing protein and finding a novel enzyme pathway from feedstock (sugar, oil) to synthesize that protein. The key idea is to use evolutionary pathways that nature has found—de novo enzymes are very hard. Ideally find some map of existing enzyme pathways to get the desired. Search databases, but also ML-based design that learns from "focused selection" after iterated modeling / testing. Enzyme design is FUCKING HARD so you get this 'lab-in-the-loop' mechanic.
- Build: Ginkgo has a big, programmable, protocol-based, modular robotics platform for biology at scle, complete with a scheduling 'OS' and train tracks to shuffle live organisms from one station to another. Usual assembly and insertion methods. Steps: chassis selection (E. coli, or beer yeast S. cerevisiae), design, fermentation, processing, QC. Want to utilize natural feedback loops where possible, e.g. yeast produces ethanol. Robotics systems have been custom-built for specific pipelines, but want a generalizable one.
- Test: "part validation" (checking intermediate steps). Evolution factors in a bunch of self-clean-up, garbage collection etc. that we have to control for / optimize for in vitro.</p>
<h2 id="history-of-folding-algos">History of folding algos</h2>
<p>Generally, can measure entropy and enthalpy; free energy is some kind of funnel, there are several local minima ie there can be multiple ways to fold, but we want to find the local minima that has a "big receptive field"</p>
<p>Proteins have hierarchical structure, with secondary, tertiary etc. <em>Homology modeling</em> assumes new proteins will be similar to old ones with similar sequences; find similar sequences with pairwise alignment or large batched multi-sequence alignment (MSA). This works because similar sequences have shared evolutionary ancestors, which then might have similar functions. </p>
<p><em>Rosetta</em> (2000) replaced fragments one at a time, and then tried to be smart about positioning the fragments relative to each other. Combination of de-novo physics with assumptions over fragments from database and statistical torsions.</p>
<p>Then was <em>molecular dynamics</em> using basic physics simulation on individual atoms and molecules. Using "state models" instead of direct Markov simulation, cluster individual short paths into "states" and construct a brief graph between clustered states. This was competitive with Rosetta for small molecules de novo.</p>
<p>Then came covariance modeling; if residues in sequences evolved together they're probably touching. Called <em>direct coupling analysis</em> Instead of direct input to modeling, this became indirect by adding as loss into a model that output score matrix. Don't need previous folded proteins like homolog; only need previous proteins. Output contacts fed into e.g. Rosetta. Came in 2010s with cheap sequencing. Helped with protein-protein modeling. Use a <code>Potts model</code> modelling log-likelihood of proteins as $h_{i}(x_{i})+J_{ij}(x_{i},x_{j})$ and then directly maximize likelihood.</p>
<p><em>Alphafold 1</em> used MSA input with a 2D CNN to generate residue contact map then gradient-based folding algo.
Then was <em>Alphafold 2</em> using
- a deep MSA
- Evoformer with 48 sequential full attention/cross attention transformer blocks between embedded MSA sequences and representation of the pair interactions
- structure module iteratively refining geometric backbone to match pair interactions representation (partly true analog representation e.g. distance, angles)
- uses MSA but all folded proteins were learned into model weights, no homolog stuff</p></p>
    </article>
</section>
<!-- endblock -->
                
                
            </main>

            
                    <!-- block preview -->
<!-- endblock -->
            

            
                    <!-- block footer -->
<footer class="pt-4 my-md-5 pt-md-5 border-top" id="component-footer">
    <div class="row">
        <div class="col-12 col-md">
                <!-- block copyright -->

    <small class="d-block mb-3">
        Made with
        <a href="https://github.com/FernandoCelmer/mkdocs-simple-blog" target="_blank" rel="noopener">
            Simple Blog for MkDocs
        </a>
    </small>

<!-- endblock -->
        </div>
    </div>
</footer>
<!-- endblock -->
            
        </div>

            <script>var base_url = '../../../../..';</script>
            <script src="../../../../../assets/js/jquery-3.3.1.slim.min.js""></script>
            <script src="../../../../../assets/js/bootstrap.bundle.min.js""></script>
            <script src="../../../../../assets/js/main.min.js""></script>
                <script src="../../../../../extra.js" defer></script>
                <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js" defer></script>
                <script src="../../../../../search/main.js" defer></script>

    </body>

</html>